File: examples/example_sim.py
'''
Simple example usage for FPsim
'''

import sciris as sc
import fpsim as fp
from fpsim import plotting as plt

# Set options
do_plot = True
pars = dict(location='senegal')
pars['n_agents'] = 500 # Small population size
pars['end_year'] = 2020 # 1961 - 2020 is the normal date range
pars['exposure_factor'] = 1.0 # Overall scale factor on probability of becoming pregnant

sc.tic()
sim = fp.Sim(pars=pars)
sim.run()

if do_plot:
    sim.plot()
    plt.plot_calib(sim)

sc.toc()
print('Done.')

File: examples/example_sim.R
# Example FPsim usage from R.
#
# Run with e.g.:
#
#    Rscript example_sim.R

# Imports
library(ggplot2)
library(reticulate)
fp <- import('fpsim')
sp <- import('fp_analyses.senegal_parameters')

# Create
pars <- sp$make_pars()
pars$n_agents <- 500 # Set population size
sim <- fp$Sim(pars=pars)

# Run
sim$run()

# Plot
df = data.frame(t=sim$results$t, pop_size=sim$results$pop_size_months)
ggplot(df, aes(x=t, y=pop_size)) + geom_point()
